freedomofpress/securedrop-workstation
# WIP Intial Qubes 4.2 Updater Integration

## Status
Work in progress. Based on 4_2-compat-deb12-base, only @deeplow's commits are relevant to this PR.

TODO:
  - [ ] Evaluate if error states are still appropriate
  - [ ] Consider if it makes sense to update even after a migration was applied (this was not the case before)

## Description of Changes

Fixes #899 . Keeps most of the existing updater logic.

Changes proposed in this pull request:
- updates templates via the CLI Qubes Updater instead of via salt
- removes functions deprecated by the new Qubes updater

## Testing
1. `make dev`
2. launch updater
3. Observe templates starting as they are updated
4. Observe app qubes getting restarted in background
5. See "successfully completed"
6. Inspect SDW-launcher logs to ensure correct updating of templates.

## Deployment

Any special considerations for deployment? Will only apply to new installs

## Checklist

### If you have made changes to the provisioning logic


> **Note**: tests don't pass due to the WIP nature of the base 4.2 branch. However, it seems that launcher-specific tests weren't passing already and it was unclear how to run the tests. 

### If you have added or removed files


### If documentation is required
